Add Generic Header component (#7061)
alphagov/govuk-frontend: packages/govuk-frontend/src/govuk/components/generic-header/README.md

> # Generic Header

Reviewer: ```suggestion
# Generic header
```

## Installation

See the [main README quick start guide](https://github.com/alphagov/govuk-frontend#quick-start) for how to install this component.

> ## Guidance and Examples

Reviewer: ```suggestion
## Guidance and examples
```

> Find out when to use the header component in your service in the [GOV.UK Design System](https://design-system.service.gov.uk/components/header).

Reviewer 2: **issue**: Wrong links and such. We probably should reword both READMEs for the header components to be explicit about "generic" and "GOV.UK".

Author: I've fixed this and amended the other header's README to be the "GOV.UK Header"

> Find out when to use the header component in your service in the [GOV.UK Design System](https://design-system.service.gov.uk/components/generic-header).

Reviewer: ```suggestion
See the Design System website for more information on [how to use the Generic header component](https://design-system.service.gov.uk/components/generic-header) in your service.
```

## Component options

> Use options to customise the appearance, content and behaviour of a component when using a macro, for example, changing the text.

Reviewer: ```suggestion
Use options to customise the appearance, content and behaviour of a component when using a macro (for example, to change the text).
```

> See [options table](https://design-system.service.gov.uk/components/generic-header/#options-generic-header-example) for details.

Reviewer: ```suggestion
See the [options table](https://design-system.service.gov.uk/components/generic-header/#options-generic-header-example) for details.
```
